website/api/controllers/entrance/signup.js

```diff
@@ -120,107 +124,135 @@ the account verification message.)`,
       return new Error(`An error occurred when trying to create a Stripe Customer for a new user with the using the email address ${newEmailAddress}. The incomplete user record has not been saved in the database, and the user will be asked to try signing up again. Full error: ${error.raw}`);
     });
 
-    // Provisioning a Fleet sandbox instance for the new user. Note: Because this is the only place where we provision Sandbox instances, We'll provision a Sandbox instance BEFORE
-    // creating the new User record. This way, if this fails, we won't save the new record to the database, and the user will see an error on the signup form asking them to try again.
-
-    const FIVE_DAYS_IN_MS = (5*24*60*60*1000);
-    // Creating an expiration JS timestamp for the Fleet sandbox instance. NOTE: We send this value to the cloud provisioner API as an ISO 8601 string.
-    let fleetSandboxExpiresAt = Date.now() + FIVE_DAYS_IN_MS;
-
-    // Creating a fleetSandboxDemoKey, this will be used for the user's password when we log them into their Sandbox instance.
-    let fleetSandboxDemoKey = await sails.helpers.strings.uuid();
-
-    // Send a POST request to the cloud provisioner API
-    let cloudProvisionerResponseData = await sails.helpers.http.post(
-      'https://sandbox.fleetdm.com/new',
-      { // Request body
-        'name': firstName + ' ' + lastName,
-        'email': newEmailAddress,
-        'password': fleetSandboxDemoKey, //« this provisioner API was originally designed to accept passwords, but rather than specifying the real plaintext password, since users always access Fleet Sandbox from their fleetdm.com account anyway, this generated demo key is used instead to avoid any confusion
-        'sandbox_expiration': new Date(fleetSandboxExpiresAt).toISOString(), // sending expiration_timestamp as an ISO string.
-      },
-      { // Request headers
-        'Authorization':sails.config.custom.cloudProvisionerSecret
+    let newUserRecord;
+
+    // If the sandbox waitlist is enabled, we'll create a record without fleetSandboxURL,fleetSandboxExpiresAt,fleetSandboxDemoKey values and with inSandboxWaitlist set to true.
+    if(sails.config.custom.fleetSandboxWaitlistEnabled === true) {
+      newUserRecord = await User.create(_.extend({
+        firstName,
+        lastName,
+        organization,
+        emailAddress: newEmailAddress,
+        signupReason,
+        password: await sails.helpers.passwords.hashPassword(password),
+        stripeCustomerId,
+        inSandboxWaitlist: true,
+        tosAcceptedByIp: this.req.ip
+      }, sails.config.custom.verifyEmailAddresses? {
+        emailProofToken: await sails.helpers.strings.random('url-friendly'),
+        emailProofTokenExpiresAt: Date.now() + sails.config.custom.emailProofTokenTTL,
+        emailStatus: 'unconfirmed'
+      }:{}))
+      .intercept('E_UNIQUE', 'emailAlreadyInUse')
+      .intercept({name: 'UsageError'}, 'invalid')
+      .fetch();
+
+    } else {
```

Comment:
Eric: These redlines are me moving it into the if/else, mostly.  The exception is the email verification logic, that we're not using anyway.